### DISCUSSION TITLE
When a task is set to failed externally, is the on_failure_callback called?

### DISCUSSION THREAD

**Discussion by picarro-pjiang** [apache/airflow#32415] [Q&A]
When I change the state of a running task to failure, it seems the on_failure_callback function not get called. Is this expected behavior? Is there a way to trigger the on_failure_callback when the a task set to failed from UI or by restful API? Thanks!

**Comment by potiuk** [comment]
See https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/callbacks.html#callbacks

> Callback functions are only invoked when the task state changes due to execution by a worker. As such, task changes set by the command line interface ([CLI](https://airflow.apache.org/docs/apache-airflow/stable/howto/usage-cli.html)) or user interface ([UI](https://airflow.apache.org/docs/apache-airflow/stable/ui.html)) do not execute callback functions.


No. There is no way to do it currently.
